src/main/include/subsystems/BaseSingleAxisSubsystem.h

```diff
@@ -0,0 +1,372 @@
+   * limit. Use RunMotorExternal if moving the motor from an external command
+   *
+   * @param speed Percentage speed
+   */
+  void RunMotorSpeed(double speed, bool ignoreEncoder = false) override {
+    if (_log)
+      Logging::logToStdOut(_prefix,
+                           "MUL IS " + std::to_string(_config.motorMultiplier),
+                           Logging::Level::VERBOSE, _ansiPrefixModifiers);
+    speed *= _config.motorMultiplier;
+    speed = std::clamp(speed, -1.0, 1.0);
+    if (_log)
+      Logging::logToStdOut(_prefix, "SPEED IS " + std::to_string(speed),
+                           Logging::Level::VERBOSE, _ansiPrefixModifiers);
+
+    bool homeState = ignoreEncoder ? AtLimitSwitchHome() : AtHome();
+    if (homeState) {
+      if (_log)
+        Logging::logToStdOut(_prefix, "AT HOME", Logging::Level::INFO,
+                             _ansiPrefixModifiers);
+      if (speed < 0) {
+        if (_log)
+          Logging::logToStdOut(_prefix,
+                               "SETTING SPEED TO: " + std::to_string(speed),
+                               Logging::Level::VERBOSE, _ansiPrefixModifiers);
+        _motor.Set(speed);
+        return;
+      }
+
+      if (_log)
+        Logging::logToStdOut(_prefix,
+                             "NOT MOVING; AT HOME" + std::to_string(speed),
+                             Logging::Level::VERBOSE, _ansiPrefixModifiers);
+
+      _motor.Set(0);
+      return;
+    }
+
+    else if (AtMax()) {
+      if (_log)
+        Logging::logToStdOut(_prefix, "AT MAX", Logging::Level::INFO);
+      if (speed > 0) {
+        if (_log)
+          Logging::logToStdOut(_prefix,
+                               "SETTING SPEED TO: " + std::to_string(speed),
+                               Logging::Level::VERBOSE, _ansiPrefixModifiers);
+        _motor.Set(speed);
+        return;
+      }
+
+      if (_log)
+        Logging::logToStdOut(_prefix,
+                             "NOT MOVING; AT MAX " + std::to_string(speed),
+                             Logging::Level::VERBOSE, _ansiPrefixModifiers);
+      _motor.Set(0);
+      return;
+    } else {
+      if (_log)
+        Logging::logToStdOut(_prefix,
+                             "SETTING SPEED TO: " + std::to_string(speed),
+                             Logging::Level::VERBOSE, _ansiPrefixModifiers);
+      _motor.Set(speed);
+    }
+  }
+
+  void RunMotorSpeedDefault(bool invertDirection = false) override {
+    RunMotorSpeed(invertDirection ? -_config.defaultMovementSpeed
+                                  : _config.defaultMovementSpeed);
+  }
+
+  /**
+   * @brief Call this one from a joystick-bound command to override current
+   * movement
+   *
+   * @param speed Percentage speed
+   */
+  void RunMotorExternal(double speed) override {
+    // TODO: constant
+    if (abs(speed) <= 0.05) {
+      if (_isMovingToPosition)
+        return; // Don't set the motor and overwrite a potential
```

Comment:
Why is the movement not overriden when speed is low but is overridden down below when speed is higher? Comment might help.

Reply:
This method is called from an external source such as a joystick. We don't want its default of ~0 to override an automated motion, but we _do_ want to take over from the automated movement if needed.